platform/mbed_error.h

```diff
@@ -60,7 +67,17 @@
 extern "C" {
 #endif
 
-void error(const char* format, ...);
+
+static inline void error(const char *format, ...) {
+#ifndef NDEBUG
+    va_list arg;
+    va_start(arg, format);
+    mbed_error_vfprintf(format, arg);
+    va_end(arg);
+#endif
+    while (1) {}
```

Comment:
mbed_die still pulls in ~2Kbytes into the base image (for gpio to flash leds?).

I've changed error to still call exit in develop/release so streams are at least flushed while developing. This is a change between develop/debug, but it may be a worthwhile tradeoff for the space savings in the general case. Thoughts?